CRM/Core/CodeGen/Util/MessageTemplates.php

```diff
@@ -0,0 +1,82 @@
+        'contribution_recurring_edit'    => ts('Contributions - Recurring Updates', ['escape' => 'sql']),
+        'pcp_notify'                    => ts('Personal Campaign Pages - Admin Notification', ['escape' => 'sql']),
+        'pcp_status_change'             => ts('Personal Campaign Pages - Supporter Status Change Notification', ['escape' => 'sql']),
+        'pcp_supporter_notify'          => ts('Personal Campaign Pages - Supporter Welcome', ['escape' => 'sql']),
+        'pcp_owner_notify'              => ts('Personal Campaign Pages - Owner Notification', ['escape' => 'sql']),
+        'payment_or_refund_notification' => ts('Additional Payment Receipt or Refund Notification', ['escape' => 'sql']),
+      ],
+      'event' => [
+        'event_offline_receipt' => ts('Events - Registration Confirmation and Receipt (off-line)', ['escape' => 'sql']),
+        'event_online_receipt'  => ts('Events - Registration Confirmation and Receipt (on-line)', ['escape' => 'sql']),
+        'event_registration_receipt'  => ts('Events - Receipt only', ['escape' => 'sql']),
+        'participant_cancelled' => ts('Events - Registration Cancellation Notice', ['escape' => 'sql']),
+        'participant_confirm'   => ts('Events - Registration Confirmation Invite', ['escape' => 'sql']),
+        'participant_expired'   => ts('Events - Pending Registration Expiration Notice', ['escape' => 'sql']),
+        'participant_transferred'   => ts('Events - Registration Transferred Notice', ['escape' => 'sql']),
+      ],
+      'friend' => [
+        'friend' => ts('Tell-a-Friend Email', ['escape' => 'sql']),
+      ],
+      'membership' => [
+        'membership_offline_receipt' => ts('Memberships - Signup and Renewal Receipts (off-line)', ['escape' => 'sql']),
+        'membership_online_receipt'  => ts('Memberships - Receipt (on-line)', ['escape' => 'sql']),
+        'membership_autorenew_cancelled' => ts('Memberships - Auto-renew Cancellation Notification', ['escape' => 'sql']),
+        'membership_autorenew_billing' => ts('Memberships - Auto-renew Billing Updates', ['escape' => 'sql']),
+      ],
+      'meta' => [
+        'test_preview' => ts('Test-drive - Receipt Header', ['escape' => 'sql']),
+      ],
+      'pledge' => [
+        'pledge_acknowledge' => ts('Pledges - Acknowledgement', ['escape' => 'sql']),
+        'pledge_reminder'    => ts('Pledges - Payment Reminder', ['escape' => 'sql']),
+      ],
+      'uf' => [
+        'uf_notify' => ts('Profiles - Admin Notification', ['escape' => 'sql']),
+      ],
+      'petition' => [
+        'petition_sign' => ts('Petition - signature added', ['escape' => 'sql']),
+        'petition_confirmation_needed' => ts('Petition - need verification', ['escape' => 'sql']),
+      ],
+    ];
+    $smarty->assign('ogNames', $ogNames);
+    $smarty->assign('ovNames', $ovNames);
+    $dir = $smarty->get_template_vars()['gencodeXmlDir'] . '/templates/message_templates/sample';
```

Comment:
It doesn't matter per se but because we merged the smarty compatibility class `getTemplateVars()` is preferred - that's the v3 version so at some point we need to swap although both work right now

Reply:
yeh it might matter more here because of https://github.com/civicrm/civicrm-core/pull/27727/files#diff-2131b62e0019c60c9029efe45018ac61ae5f687d791013825679af5f9bdf28edR17 doesn't use the compatibility class

Reply:
ok - we might as well punt switching that one then!